src/BufferCopy/RDRAMtoColorBuffer.cpp

```diff
@@ -221,6 +238,22 @@ void RDRAMtoColorBuffer::copyFromRDRAM(u32 _address, bool _bCFB)
 			bCopy = _copyPixelsFromRdram<u32>(address, m_vecAddress, dst, RGBA32ToABGR32, 0, width, height, _bCFB);
 	}
 
+	//Convert integer format to float
+	if (fbTexFormats.colorType == datatype::FLOAT) {
+		f32* floatData = reinterpret_cast<f32*>(convertedDataPtr);
+		u8* byteData = dstData.get();
+		int widthPixels = width*4;
+		for (unsigned int heightIndex = 0; heightIndex < height; ++heightIndex) {
+			for (unsigned int widthIndex = 0; widthIndex < widthPixels; ++widthIndex) {
+				u8& src = *(byteData + heightIndex*widthPixels + widthIndex);
+				float& dst = *(floatData + heightIndex*widthPixels + widthIndex);
+				dst = src/255.0f;
+			}
+		}
+	}
+
+	std::copy_n(convertedDataPtr, gpuDataSize, ptr);
```

Comment:
What I'm finding, at least for mobile, is that it seems faster to do one big copy instead of many small copies from GPU memory. I didn't test the performance difference for writing to GPU memory, but I assumed that it would be similar. That's why I did that here.

What do you think?